api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -159,11 +160,39 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 						mLog.Errorf("Failed to unmarshal instance state for catalog %s: %s", name, err.Error())
 						return outputs, false, err
 					}
-					// If inner instace defines a display name, use it as the name
-					if instanceState.Spec.DisplayName != "" {
-						instanceState.ObjectMeta.Name = instanceState.Spec.DisplayName
+
+					if instanceState.ObjectMeta.Name == "" {
+						mLog.Errorf("Instance name is empty: catalog - %s", name)
+						return outputs, false, v1alpha2.NewCOAError(nil, fmt.Sprintf("Empty instance name: catalog - %s", name), v1alpha2.BadRequest)
+					}
+
+					instanceName := instanceState.ObjectMeta.Name
+					parts := strings.Split(instanceName, ":")
+					if len(parts) == 2 {
+						instanceState.Spec.RootResource = parts[0]
+						instanceState.Spec.Version = parts[1]
+					} else {
+						mLog.Errorf("Instance name is invalid: instance - %s, catalog - %s", instanceName, name)
+						return outputs, false, v1alpha2.NewCOAError(nil, fmt.Sprintf("Instance name is invalid: catalog - %s", name), v1alpha2.BadRequest)
+					}
+
+					mLog.Debugf("  P (Materialize Processor): check instance contains %v, namespace %s", instanceState.ObjectMeta.Name, namespace)
+					_, err := i.ApiClient.GetInstanceContainer(ctx, instanceState.Spec.RootResource, namespace, i.Config.User, i.Config.Password)
+					if err != nil && strings.Contains(err.Error(), constants.NotFound) {
```

Comment:
I don't think we should have different behaviors for different providers. If materialize provider creates containers if containers don't exist, create provider should also do the same way.

Reply:
Sync offline. Users should create container objects before using create stage to create a versioned object.